internal/app/machined/pkg/controllers/block/lvm.go

```diff
@@ -0,0 +1,132 @@
+
+package block
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/optional"
+	"github.com/siderolabs/go-cmd/pkg/cmd"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/resources/block"
+	runtimeres "github.com/siderolabs/talos/pkg/machinery/resources/runtime"
+	"github.com/siderolabs/talos/pkg/machinery/resources/v1alpha1"
+)
+
+// LVMActivationController activates LVM volumes when they are discovered by the block.DiscoveryController.
+type LVMActivationController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *LVMActivationController) Name() string {
+	return "block.LVMActivationController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *LVMActivationController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: v1alpha1.NamespaceName,
+			Type:      runtimeres.MountStatusType,
+			ID:        optional.Some(constants.EphemeralPartitionLabel),
+			Kind:      controller.InputWeak,
+		},
+		{
+			Namespace: block.NamespaceName,
+			Type:      block.DiscoveredVolumeType,
+			Kind:      controller.InputWeak,
+		},
+	}
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *LVMActivationController) Outputs() []controller.Output {
+	return nil
+}
+
+// Run implements controller.Controller interface.
+//
+//nolint:gocyclo
+func (ctrl *LVMActivationController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
+	touchedIDs := make(map[string]struct{})
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-r.EventCh():
+			if _, err := safe.ReaderGetByID[*runtimeres.MountStatus](ctx, r, constants.EphemeralPartitionLabel); err != nil {
+				if state.IsNotFoundError(err) {
+					// wait for the mount status to be available
+					continue
+				}
+
+				return fmt.Errorf("failed to get mount status: %w", err)
+			}
+		}
+
+		discoveredVolumes, err := safe.ReaderListAll[*block.DiscoveredVolume](ctx, r)
+		if err != nil && !state.IsNotFoundError(err) {
+			return fmt.Errorf("failed to list discovered volumes: %w", err)
+		}
+
+		for iterator := discoveredVolumes.Iterator(); iterator.Next(); {
+			if iterator.Value().TypedSpec().Name != "lvm2-pv" {
+				continue
+			}
+
+			// first we check if all associated volumes are available
+			// https://man7.org/linux/man-pages/man7/lvmautoactivation.7.html
+			stdOut, err := cmd.RunContext(ctx,
+				"/sbin/lvm",
+				"pvscan",
+				"--cache",
+				"--verbose",
+				"--listvg",
+				"--checkcomplete",
+				"--vgonline",
+				"--autoactivation",
+				"event",
+				"--udevoutput",
+				iterator.Value().TypedSpec().DevPath,
+			)
+			if err != nil {
+				return fmt.Errorf("failed to check LVM volume availability: %w", err)
```

Comment:
should we log the error and continue looking for other volumes here?

Reply:
or grab all errors and report the error in the end, as this will block all LVMs if one of them fails